OpenRA.Mods.Common/Traits/Player/ClassicProductionQueue.cs

```diff
@@ -46,39 +46,47 @@ public ClassicProductionQueue(ActorInitializer init, ClassicProductionQueueInfo
 			this.info = info;
 		}
 
-		[Sync] bool isActive = false;
-
 		protected override void Tick(Actor self)
 		{
 			// PERF: Avoid LINQ.
-			isActive = false;
+			Enabled = false;
+			var isActive = false;
 			foreach (var x in self.World.ActorsWithTrait<Production>())
 			{
-				if (x.Actor.Owner == self.Owner && x.Trait.Info.Produces.Contains(Info.Type))
-				{
-					isActive = true;
-					break;
-				}
+				if (x.Trait.IsTraitDisabled)
+					continue;
+
+				if (x.Actor.Owner != self.Owner || !x.Trait.Info.Produces.Contains(Info.Type))
+					continue;
+
+				if (!Enabled)
+					Enabled = ValidFaction;
+
+				if (!isActive)
+					isActive = !x.Trait.IsTraitPaused;
 			}
 
-			base.Tick(self);
+			if (!Enabled)
+				ClearQueue();
+
+			TickInner(self, !isActive);
 		}
 
 		public override IEnumerable<ActorInfo> AllItems()
 		{
-			return isActive ? base.AllItems() : NoItems;
+			return Enabled ? base.AllItems() : NoItems;
 		}
 
 		public override IEnumerable<ActorInfo> BuildableItems()
 		{
-			return isActive ? base.BuildableItems() : NoItems;
+			return Enabled ? base.BuildableItems() : NoItems;
 		}
 
 		public override TraitPair<Production> MostLikelyProducer()
 		{
 			return self.World.ActorsWithTrait<Production>()
 				.Where(x => x.Actor.Owner == self.Owner
-					&& x.Trait.Info.Produces.Contains(Info.Type))
+					&& !x.Trait.IsTraitDisabled && !x.Trait.IsTraitPaused && x.Trait.Info.Produces.Contains(Info.Type))
 				.OrderByDescending(x => x.Actor.IsPrimaryBuilding())
 				.ThenByDescending(x => x.Actor.ActorID)
 				.FirstOrDefault();
```

Comment:
Please change this to
```csharp
			var productionActors = self.World.ActorsWithTrait<Production>()
				.Where(x => x.Actor.Owner == self.Owner
					&& !x.Trait.IsTraitDisabled && x.Trait.Info.Produces.Contains(Info.Type))
				.OrderByDescending(x => x.Actor.IsPrimaryBuilding())
				.ThenByDescending(x => x.Actor.ActorID)
				.ToList();

			var unpaused = productionActors.FirstOrDefault(a => !a.Trait.IsTraitPaused);
			return unpaused.Trait != null ? unpaused : productionActors.FirstOrDefault();
```

to fix the sidebar glitches when a production structure in TS is EMPed.